src/components/ListResources.vue

```diff
@@ -0,0 +1,148 @@
+<template>
+  <RecycleScroller
+    v-if="resources.length"
+    v-slot="{ item: rowItem, index, active }"
+    :key="resources.length"
+    :items="resourcesSorted"
+    :item-size="68"
+  >
+    <div
+      :key="rowItem.viewId"
+      :data-is-visible="active"
+      :class="rowClasses(rowItem)"
+      @click="selectResource(rowItem)"
+    >
+      <oc-grid
+        :id="'file-row-' + rowItem.id"
+        :ref="index === 0 ? 'firstRow' : null"
+        gutter="small"
+        flex
+        class="uk-padding-small oc-border-top"
+        @click="selectResource(rowItem)"
+      >
+        <div v-if="!isLocationPicker">
+          <oc-checkbox
+            class="uk-margin-small-left"
+            :value="isResourceSelected(rowItem)"
+            :label="selectCheckboxLabel(rowItem.name)"
+            :hide-label="true"
+            @click.stop
+            @change.native="selectResource(rowItem)"
+          />
+        </div>
+        <resource class="uk-width-auto" :item="rowItem" @navigate="openFolder" />
+      </oc-grid>
+    </div>
+  </RecycleScroller>
+</template>
+
+<script>
+import { RecycleScroller } from 'vue-virtual-scroller'
+import { sortByName } from '../helpers/sort'
+import Resource from './Resource.vue'
+
+export default {
+  name: 'ListResources',
+
+  components: {
+    RecycleScroller,
+    Resource
+  },
+
+  props: {
+    resources: {
+      type: Array,
+      required: true
+    },
+    isLocationPicker: {
+      type: Boolean,
+      required: false,
+      default: false
+    },
+    currentFolder: {
+      type: Object,
+      required: true
+    }
+  },
+
+  data: () => ({
+    selectedResources: []
+  }),
+
+  computed: {
+    resourcesSorted() {
+      return this.sortResources(this.resources)
+    }
+  },
+
+  methods: {
+    openFolder(path) {
+      this.$emit('openFolder', path)
+    },
+
+    selectResource(resource) {
```

Comment:
is this more like toggling the selection ? there's code that looks like unselecting

Reply:
You're right. I changed the name to `toggleResourceSelection`